### PR TITLE
espressif: esp32 & esp32s2: drivers: clock_control

### PR DESCRIPTION
**Pull request by glaubermaroto**
This PR includes:

* ESP32's clock_control driver refactoring (to better benefit from Espressif's HAL support)
* ESP32-S2's clock_control driver support

### CODE REVIEW

**Review by uLipe — APPROVED** [review]

**Comment by sylvioalves** [comment]
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/38540

**Review by sylvioalves — APPROVED** [review]

**Comment by glaubermaroto** [comment]
The last push was just to add code ownership into the `CODEOWNERS` file.

**Comment by glaubermaroto** [comment]
Hi folks, can we move on with this PR or are we still waiting on the release version?

**Pull request merged**
